crates/astria-core/src/sequencerblock/v1/block.rs

```diff
@@ -772,6 +772,10 @@ impl SequencerBlock {
     /// # Panics
     ///
     /// - if a rollup data merkle proof cannot be constructed.
+    #[expect(
```

Comment:
Refactor this into a builder.

Needing all the arguments is not a good reason to ignore the lint.